foreman/data_refinery_foreman/surveyor/test_sra.py

```diff
@@ -112,6 +112,34 @@ def test_srp_survey(self, mock_send_task):
         self.assertEqual(experiment.alternate_accession_code, None)
         self.assertEqual(len(samples), 9)
 
+    @vcr.use_cassette("/home/user/data_store/cassettes/surveyor.sra.survey_file_report.yaml")
+    @patch("data_refinery_foreman.surveyor.external_source.send_job")
+    def test_survey_file_report(self, mock_send_job):
+        """A slightly harder test of the SRA surveyor.
+        """
+        survey_job = SurveyJob(source_type="SRA")
+        survey_job.save()
+        key_value_pair = SurveyJobKeyValue(
+            survey_job=survey_job, key="experiment_accession_code", value="SRP048683"
+        )
+        key_value_pair.save()
+
+        sra_surveyor = SraSurveyor(survey_job)
+        experiment, samples = sra_surveyor.discover_experiment_and_samples()
+
+        self.assertEqual(experiment.accession_code, "SRP048683")
+        self.assertEqual(len(samples), 12)
+
+        # Just check one file for one sample's expected file size/md5
+        for sample in samples:
+            if sample.accession_code == "SRR1603661":
+                for original_file in sample.original_files.all():
```

Comment:
I think we should also assert that `set(o.source_filename for o in sample.original_files.all())` is equal to what we expect, because if there isn't an original file with source filename `SRR1603661_1.fastq.gz` then the check succeeds when it should not.